### ISSUE TITLE
4.0.1 Crash on Insert Disk

### ISSUE THREAD

**Issue by bluecursor** [dirkwhoffmann/virtualc64#659]
Exception Type:        EXC_BAD_INSTRUCTION (SIGILL)
Exception Codes:       0x0000000000000001, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Illegal instruction: 4
Termination Reason:    Namespace SIGNAL, Code 0x4
Terminating Process:   exc handler [93201]

Application Specific Information:
Performing @selector(insertAction:) from sender NSButton 0x7fa15bdb3400
Fatal error: file VirtualC64/ImportDialog.swift, line 342
[chucks-disk-cracker.zip](https://github.com/dirkwhoffmann/virtualc64/files/6251171/chucks-disk-cracker.zip)

Works Fine in V3.4

**Comment by dirkwhoffmann** [comment]
Error happens in `FSDevice::importVolume`: 

```
    // Only proceed if the buffer size matches
    if (blocks.size() * 256 != size) {
        warn("BUFFER SIZE MISMATCH (%lu %lu)\n", blocks.size(), blocks.size() * 256);
        if (err) *err = ERROR_FS_WRONG_CAPACITY;
        return false;
    }
```

The importer thinks the buffer size is wrong, because it's not a multiple of 256. However, the buffer size is fine. It's not a multiple of 256 because the D64 file contains error codes at the end. It's a D64 file of type `D64_683_SECTORS_ECC`:

```
    // D64 files come in six different sizes
    static const usize D64_683_SECTORS     = 174848;
    static const usize D64_683_SECTORS_ECC = 175531;
    static const usize D64_768_SECTORS     = 196608;
    static const usize D64_768_SECTORS_ECC = 197376;
    static const usize D64_802_SECTORS     = 205312;
    static const usize D64_802_SECTORS_ECC = 206114;
```

**Comment by dirkwhoffmann** [comment]
Should be fixed in v4.1.
